### ISSUE TITLE
JdbcDbClient uses get/setString for binary data

### ISSUE THREAD

**Issue by vlsi** [brianfrankcooper/YCSB#1089]
TL;DR: does it make sense to replace `setString`/`getString` with `setBytes`/`getBytes`?

Current table structure for Jdbc is `FIELDN TEXT`, however the stored data is binary. It causes `String <-> byte[]` conversions.

JDBC has multiple ways to upload binary:
`void setBinaryStream(int parameterIndex, java.io.InputStream x)`
`void setBinaryStream(int parameterIndex, java.io.InputStream x, int length)`
`void setBytes(int parameterIndex, byte x[])`

It might easily be the case that `setBytes(int, byte[])` turns out to be more effective than `setString(int, byteIterator.toString())`

On top of that, certain JDBC drivers might implement extensions to stream bytes from a source without intermediate `byte[]` copy (see https://github.com/pgjdbc/pgjdbc/pull/953 )

On the read side, JDBC has
`byte[] getBytes(int columnIndex)`
`java.io.InputStream getBinaryStream(int columnIndex)`

Read part can have extensions as well via `public <T> T getObject(int columnIndex, Class<T> type)`.

See 
https://github.com/brianfrankcooper/YCSB/blob/d6e57c3f575914c32556ca4eb5802073681b288c/jdbc/src/main/java/com/yahoo/ycsb/db/JdbcDBClient.java#L385
https://github.com/brianfrankcooper/YCSB/blob/d6e57c3f575914c32556ca4eb5802073681b288c/jdbc/src/main/java/com/yahoo/ycsb/db/JdbcDBClient.java#L413

**Comment by busbey** [comment]
yeah this would be a nice improvement.
